node-util/sbin/oo-init-quota

```diff
@@ -49,12 +49,12 @@ function get_mountpoint() {
 
 # find the filesystem type above a given file or directory
 function get_filesystem_type() {
-    egrep "\s+$1\s+" /etc/fstab | awk '{ print $3 }'
+    egrep "^[^#].+\s+$1\s+" /etc/fstab | awk '{ print $3 }'
 }
 
 # get the fstab options for a given mount point
 function get_fstab_mount_options() {
-    cat /etc/fstab | tr -s ' ' | grep " $1 "| awk '{print $4;}'
+    egrep "^[^#].+\s+$1\s+" /etc/fstab | awk '{ print $4 }'
 }
```

Comment:
Since this regex is duplicated it might be worthwhile to place it in a variable and add some comments explaining what it's handling.  I can't say I understand the `[^#]` part.  Are there lines that begin with a literal `^` in `/etc/fstab`?